modules/cas_cache/volume/vol_block_dev_top.c

```diff
@@ -304,6 +308,7 @@ static void blkdev_handle_data(struct bd_object *bvol, struct bio *bio)
 			CAS_BIO_OP_FLAGS_FORMAT "\n",  CAS_BIO_OP_FLAGS(bio));
 		CAS_BIO_ENDIO(bio, CAS_BIO_BISIZE(bio),
 				CAS_ERRNO_TO_BLK_STS(-EINVAL));
+		cas_blk_queue_exit(bvol);
```

Comment:
I just realized - if it's always used together with `CAS_BIO_ENDIO` couldn't we make this workaround a part of `CAS_BIO_ENDIO`?